internal/config/config.go

```diff
@@ -127,6 +128,94 @@ func defaultExists(key string) bool {
 	return false
 }
 
+// AuthConfig is used for interacting with some persistent configuration for gh,
+// with knowledge on how to access encrypted storage when neccesarry.
+// Behavior is scoped to authentication specific tasks.
+type AuthConfig struct {
+	cfg           *ghConfig.Config
+	hostsOverride func() []string
+	tokenOverride func(string) (string, string)
```

Comment:
I tried a new approach here to have the overrides live inside the struct rather than create an interface that can be mocked out. Any function that calls a `ghAuth` can have an override.